editor/src/clj/editor/outline.clj

```diff
@@ -109,13 +111,12 @@
                                          (when-some [tx-attach-fn (:tx-attach-fn req)]
                                            (let [target-id (g/override-root (:node-id item))
                                                  tx-data (tx-attach-fn target-id child-id)]
-                                             (keep (fn [tx-step]
-                                                     (when (= :connect (:type tx-step))
-                                                       (let [src-serial-id (node-id->serial-id (:source-id tx-step))
-                                                             tgt-serial-id (node-id->serial-id (:target-id tx-step))]
-                                                         (when (and src-serial-id tgt-serial-id)
-                                                           [src-serial-id (:source-label tx-step) tgt-serial-id (:target-label tx-step)]))))
-                                                   (flatten tx-data)))))))
+                                             (map (fn [arc]
+                                                    (let [src-serial-id (node-id->serial-id (gt/source-id arc))
+                                                          tgt-serial-id (node-id->serial-id (gt/target-id arc))]
+                                                      (when (and src-serial-id tgt-serial-id)
+                                                        [src-serial-id (gt/source-label arc) tgt-serial-id (gt/target-label arc)])))
```

Comment:
why use map instead of keep if fn can still return nil?

Reply:
Whoops! Yep, this still needs to use `keep`. Fixing.